client/client.go

```diff
@@ -67,6 +69,17 @@ func Configure(logger hclog.Logger, config interface{}) (schema.ClientMeta, diag
 		}
 	}
 
+	logger.Info("Trying to authenticate via CLI (azidentity)")
+	var azCred azcore.TokenCredential
+	azCred, err = azidentity.NewAzureCLICredential(nil)
+	if err != nil {
+		logger.Info("Trying to authenticate via environment variables (azidentity)")
+		azCred, err = azidentity.NewEnvironmentCredential(nil)
+		if err != nil {
+			return nil, diag.FromError(err, diag.USER)
+		}
+	}
+
```

Comment:
How will this change impact users that are grabbing subscriptions already? I ask because I am not sure of the different ways of sourcing Azure credentials

Reply:
Good question, but I don't think they will be impacted. I tried to keep the code block here equivalent to the old-SDK [one just above it](https://github.com/cloudquery/cq-provider-azure/pull/412/files#diff-bd3a55a72186f59e2e63efb4951573b2f9e4a7cc98086e922b0859f8ccc1dd09R62-R70).

So like before, we try to use CLI credentials first, and if that fails, we try to use environment credentials.